app/models/concerns/hyku_addons/account_behavior.rb

```diff
@@ -31,6 +37,19 @@ def datacite_endpoint
 
     private
 
+      def validate_email_format
```

Comment:
It's not clear to me if this method is intended for checking if `email_format` is in the proper format or is intended to do data manipulation.  Same for `validate_contact_emails`.  I would expect validation methods to only be checking for proper format and data manipulation to be happening in the controller or form or setter methods.

Reply:
There is a lot of validation logic into the Account model that is not intrinsic to the Account itself, but dependent on the context (for example, should an account always validate the settings email format, or rather only do it when you are updating settings?). 

It looks like we could extract all the settings functionality into a Form object as Chris says. https://codeclimate.com/blog/7-ways-to-decompose-fat-activerecord-models/

Reply:
I see what you are saying @cjcolvar let me try something else